README.md

```diff
@@ -1,207 +1,249 @@
-- [JSON Schema](json-schema/schema.json)
+- [JSON Schema](json-schema/schema.json) TODO update
 - [Changelog](./CHANGELOG.md)
 
 ## Item Properties and Collection Fields
 
-| Field Name       | Type                                        | Description                                                            |
-|------------------|---------------------------------------------|------------------------------------------------------------------------|
-| dlm:data         | [Data Object](#data-object)                 | Describes the EO data compatible with the model.                       |
-| dlm:inputs       | [Inputs Object](#inputs-object)             | Describes the transformation between the EO data and the model inputs. |
-| dlm:architecture | [Architecture Object](#architecture-object) | Describes the model architecture.                                      |
-| dlm:runtime      | [Runtime Object](#runtime-object)           | Describes the runtime environments to run the model (inference).       |
-| dlm:outputs      | [Outputs Object](#outputs-object)           | Describes each model output and how to interpret it.                   |
+| Field Name                  | Type                                          | Description                                                                                                                                                                                                                                                            |
+|-----------------------------|-----------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| mlm:name                    | string                                        | **REQUIRED.** A unique name for the model. This should include but be distinct from simply naming the model architecture. If there is a publication or other published work related to the model, use the official name of the model.                                  |
+| mlm:task                    | [Task Enum](#task-enum)                       | **REQUIRED.** Specifies the primary Machine Learning task for which the output can be used for. If there are multi-modal outputs, specify the primary task and specify each task in the [Model Output Object](#model-output-object).                                   |
+| mlm:framework               | string                                        | **REQUIRED.** Framework used to train the model (ex: PyTorch, TensorFlow).                                                                                                                                                                                             |
```

Comment:
Although the spec should remain generic about specific names to allow extending to new definitions, a lot of these are very common. We should probably provide an "official list" of known frameworks (which we can gradually extend upon request) just to increase the chances of unifying the properties across created STAC Item entries. 

I suggest the JSON-schema should define the following (corresponding information in the README) to get the best of both worlds (open to other suggestions/alternatives): 

```yaml
fields:
  mlm:framework:
    anyOf: 
    - type: string
      $comment: Add more entries here as needed, an repeat them in the README.
      enum: 
      - PyTorch
      - TensorFlow
      - Scikit-learn
      - Huggingface
      - PyMC
      - ONNX
      - JAX
      - MXNet
      - Keras
      - Caffe
      - Weka
    - type: string
      $comment: |
        Any other framework name to allow extension.
        Enum names should be prefered when possible to allow better portability.
```